packages/mobile/src/exchange/saga.ts

```diff
@@ -139,8 +140,11 @@ export function* doFetchExchangeRate(action: FetchExchangeRateAction) {
         dollarMaker: dollarMakerExchangeRate.toString(),
       })
     )
+    ValoraAnalytics.track(CeloExchangeEvents.celo_fetch_exchange_rate_complete)
```

Comment:
If not already tracked it would be helpful to track the exchange rate that was fetched and the amount it was estimated with to provide us with data in addressing the UX issues around the amount-dependent exchange rate

Reply:
We have the below check. Does this work?

`    // Ensure the user gets makerAmount at least as good as displayed (rounded to EXCHANGE_DIFFERENCE_TOLERATED)
    const minimumTakerAmount = BigNumber.maximum(
      getTakerAmount(makerAmount, exchangeRate).minus(EXCHANGE_DIFFERENCE_TOLERATED),
      0
    )
    const updatedTakerAmount = getTakerAmount(makerAmount, updatedExchangeRate)
    if (minimumTakerAmount.isGreaterThan(updatedTakerAmount)) {
      ValoraAnalytics.track(CeloExchangeEvents.celo_exchange_error, {
        error: `Not receiving enough ${makerToken}. Expected ${minimumTakerAmount} but received ${updatedTakerAmount.toString()}`,
      })
      Logger.error(
        TAG,
        `Not receiving enough ${makerToken} due to change in exchange rate. Exchange failed.`
      )
      yield put(showError(ErrorMessages.EXCHANGE_RATE_CHANGE))
      return
    }`